### DISCUSSION TITLE
firstValueFrom and lastValueFrom defaultValue type

### DISCUSSION THREAD

**Discussion by jakovljevic-mladen** [ReactiveX/rxjs#6278] [Ideas / Feature request]
Hi all. I'd like to discuss the type of the `defaultValue` parameters from `FirstValueFromConfig` and `LastValueFromConfig` interfaces.

Based on [this comment](https://github.com/ReactiveX/rxjs/pull/6204#pullrequestreview-631981668), there is a strong possibility that a new `signal` property will be added to these interfaces. Most probably, `defaultValue` will become optional so users don't have to provide it if they only want to provide an abortion signal, like this: `firstValueFrom(source$, { signal: ac.signal });`.

In that case, it won't be possible to allow promises to resolve with `undefined` instead of a promise erroring with an `EmptyError` if nothing has been sent through the stream. With current set up, it is possible to write this: `firstValueFrom(source, { defaultValue: undefined });` and if `source` is `EMPTY`, it will resolve a promise with `undefined`.

I'd like to propose `defaultValue` to be renamed to `default` and to become an object with a `value` property. If `firstValueFrom` or `lastValueFrom` receive an object, they will emit the default value from a `value` property of that object.

So, `firstValueFrom(EMPTY, { signal: ac.signal });` would reject a promise with `EmptyError`.

But, `firstValueFrom(EMPTY, { default: {}, signal: ac.signal });` (or `firstValueFrom(EMPTY, { default: { value: undefined }, signal: ac.signal });`) would resolve a promise with `undefined`.

**Comment by cartant** [comment]
If/when `signal` is added to the configuration object, I think we can make `defaultValue` optional and can test for the absence of a `defaultValue` property in the configuration object using `hasOwnProperty('defaultValue')`. If `defaultValue` is absent and the source completes without emitting a next notification, the implementation can throw.

**Reply by jakovljevic-mladen** [reply]
Oh, yeah, I keep forgetting about this handful JS feature. Thanks Nicholas. I guess this can be closed (if that's possible with discussions).

**Reply by cartant** [reply]
> I guess this can be closed

I don't think it's possible. 🙂

**Reply by kwonoj** [reply]
I think we can mark accepted answer and lock instead. Surprised there's no close in discussions.

**Reply by jakovljevic-mladen** [reply]
I can't seem to find _accept_ button either. Only upvote...
